pkg/update/update.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+
+	"github.com/nitrictech/cli/pkg/utils"
+)
+
+var (
+	shouldShowWarning bool
+	CurrentVersion    string
+	LatestVersion     string
+)
+
+func CheckOutdated() {
+	skip, err := shouldSkip()
+	if err != nil {
+		cobra.CheckErr(err)
+	}
+
+	if skip {
+		return
+	}
+
+	CurrentVersion = strings.TrimPrefix(utils.Version, "v")
+
+	owner := "nitrictech"
+	repo := "cli"
+
+	apiURL := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases/latest", owner, repo)
+
+	response, err := http.Get(apiURL)
+	if err != nil {
+		// if there is an error due to being offline, timeout or rate limit. Skip check.
+		return
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		cobra.CheckErr(err)
+	}
+
+	var releaseInfo struct {
+		TagName string `json:"tag_name"`
+	}
+
+	err = json.Unmarshal(body, &releaseInfo)
+	if err != nil {
+		cobra.CheckErr(err)
+	}
+
+	LatestVersion = strings.TrimPrefix(releaseInfo.TagName, "v")
+
+	shouldShowWarning = CurrentVersion < LatestVersion
+}
+
+func ShowWarning() {
+	if shouldShowWarning {
+		msg := fmt.Sprintf(`A new version of Nitric is available. To upgrade from version '%s' to '%s'`, CurrentVersion, LatestVersion)
+		msg += ", visit https://nitric.io/docs/installation for instructions and release notes.\n"
+
+		pterm.Warning.Printf(msg)
+
+		// only show warning once per run
+		shouldShowWarning = false
+	}
+}
+
+func cachePath() string {
+	return filepath.Join(utils.NitricHomeDir(), ".cached-last-version-check")
+}
+
+func shouldSkip() (bool, error) {
+	fileInfo, err := os.Stat(cachePath())
+	if err != nil {
+		// no file, create and return false
+		err = updateFile()
+
+		return false, err
+	}
+
+	if time.Now().After(fileInfo.ModTime().Add(24 * time.Hour)) {
```

Comment:
I think it's fine to show the version warning every time. Maybe just update the cache every day?